tests/resources/test_gemd_resource.py

```diff
@@ -179,6 +181,13 @@ def test_register_all(gemd_collection):
         parameter_template: ParameterTemplateCollection,
         condition_template: ConditionTemplateCollection
     }
+
+    for obj in expected:
+        assert len(obj.uids) == 0  # All should be without ids
+
+    # dry_run should pass for all objects and shouldn't mutate the objects
+    gemd_collection.register_all(expected.keys(), dry_run=True)
```

Comment:
NB for now. but this batch of tests would be a perfect example of things that could be improved using the fake collections infrastructure we started building.

I'm making a card for it, and add a TODO for it.
```suggestion
    # TODO(PLA-8368): Clean up test code using fake modules for a more stable interface
    gemd_collection.register_all(expected.keys(), dry_run=True)
```